api/src/main/java/org/pmiops/workbench/api/ConceptsController.java

```diff
@@ -98,13 +104,35 @@ public ConceptsController(ConceptService conceptService, WorkspaceService worksp
       throw new BadRequestException("Query must be non-whitespace");
     }
 
+    List<Long> conceptCodeIdMatchConcepts = new ArrayList<>();
+    conceptCodeIdMatchConcepts = conceptDao.findConceptByCode(request.getQuery());
+    try{
+      Long conceptIdQuery = Long.parseLong(request.getQuery());
+      conceptCodeIdMatchConcepts.addAll(conceptDao.findConceptById(conceptIdQuery));
+    }catch(NumberFormatException nfe){
+
+    }
+    List<Long> synonymConceptIds = new ArrayList<>();
+    if(conceptCodeIdMatchConcepts.size() == 0) {
+        synonymConceptIds = conceptDao.findConceptByNameOrSynonymName(ConceptService.modifyMultipleMatchKeyword(request.getQuery()));
```

Comment:
Depending on what the search query is, this could return a whole bunch of concepts -- too much to fit in memory and pass into an IN clause subsequently here. Have you tried testing this with some searches for strings that appear in a whole bunch of concepts? 

I believe it should be possible to make searchConcepts join to the synonym table and match on name at the same time as filtering by vocabulary, domain, etc. and sorting by prevalence. Then the database can do the join itself, which is more efficient and scalable than us doing it this way.

Reply:
Changed back and added concept synonym match query filter in there.